dds/DCPS/Serializer.h

```diff
@@ -669,25 +673,51 @@ class OpenDDS_Dcps_Export Serializer {
 
   void set_construction_status(ConstructionStatus cs);
 
+  struct OpenDDS_Dcps_Export ScopedAlignmentContext {
+    explicit ScopedAlignmentContext(Serializer& ser);
+    virtual ~ScopedAlignmentContext() { restore(ser_); }
+
+    void restore(Serializer& ser) const;
+
+    Serializer& ser_;
+    const size_t max_align_;
+    const size_t start_rpos_;
+    const size_t rblock_;
+    const size_t start_wpos_;
+    const size_t wblock_;
+  };
+
+  template <typename T>
+  bool peek_helper(ACE_Message_Block* const block, T& t)
+  {
+    bool result = false;
+    char* const rd_ptr = block->rd_ptr();
+    if (block->cont()) {
+      result = peek_helper(block->cont(), t);
```

Comment:
This makes the complexity of the peek `O(n)` for n message blocks.
Since we expect this to be mostly used for a peek of a basic type, should those have a specialization/overload so that they only need to look `max_padding + n_bytes` ahead?